sound/soc/sof/ipc4-loader.c

```diff
@@ -66,8 +66,9 @@ static size_t sof_ipc4_fw_parse_ext_man(struct snd_sof_dev *sdev)
 	dev_dbg(sdev->dev, "Firmware name: %s, header length: %u, module count: %u\n",
 		fw_header->name, fw_header->len, fw_header->num_module_entries);
 
+	/* allocate one more fw_module for pipeline */
```

Comment:
@bardliao I think this step is unnecessary. You can add a case SOF_IPC4_FW_CFG_MAX_PPL_COUNT to sof_ipc4_query_fw_configuration() to query the max pipeline count supported by the FW and simply use that instead of adding a new FW module.